### PR TITLE
Replace hpack with a standard cabal file

### LINKED ISSUE #111: Get rid of hpack
From my perspective, `hpack` is causing more problems than it solves.

With common stanzas, the repetetiveness of the cabal file format is much reduced, and I keep having to remember to run hpack when I'm building the project. It's also been causing some intermittent CI issues.

How do you all feel about replacing it with a `.cabal` file with a common stanza?

### PR DESCRIPTION
**Pull request by david-christiansen**
Fixes #111. This touches Travis, so I want to make sure that passes before I merge :)

### CODE REVIEW

**Comment by david-christiansen** [comment]
The tests pass with GHC 8.6.5, 8.8.4, and 8.10.2 locally, and Stack does its thing as well for me. I'll wait for Travis to be happy, then mark this as review-ready.

**Comment by david-christiansen** [comment]
The PR ends up also increasing the Travis build matrix. I had to mess with the Travis file anyway, so I figured I'd add a newer GHC to the mix while I was at it.

File: klister.cabal [hunk 1]
```diff
@@ -0,0 +1,120 @@
+cabal-version: 2.2
+
+name:           klister
```
Comment (langston-barrett) [inline klister.cabal:3]: nit: In general, I prefer non-aligned code. Consider what the diff would look like if we added the famous `really-long-cabal-field-everything-needs-to-be-reindented-now: true` field to this config.

File: klister.cabal [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    megaparsec           >= 7.0.5 && < 9.1,
+    mtl                  >= 2.2.2 && < 2.3,
+    prettyprinter        >= 1.2 && < 1.8,
+    text                 ^>= 1.2,
+    transformers         ^>= 0.5
+
+library
+  import: deps
+  exposed-modules:
+      Alpha
```
Comment (langston-barrett) [inline klister.cabal:41]: Indentation here is 4 but in `common deps`/`build-depends:` it's 2. Any reason for the discrepancy?

**Review by langston-barrett — APPROVED** [review]
Two tiny nits but nothing that should hold up merging.

**Review by gelisam — APPROVED** [review]

**Comment by gelisam** [comment]
Merging so I can restart the other PRs which failed because travis couldn't download hpack. We can address the nitpicks in a later PR if needed.

**Pull request merged**